test/app-luatest/http_client_test.lua

```diff
@@ -654,3 +654,20 @@ g.test_concurrent = function(cg)
     t.assert(ok_sockets_added, "free sockets")
     t.assert(ok_active, "no active requests")
 end
+
+g.test_auto_encode_body = function(cg)
+    local url = cg.url
+    local opts = table.deepcopy(cg.opts)
+    local body = {
+        Moscow = 1,
+        Pushkino = 2,
+        Mytischi = 2,
+    }
+    opts.headers = {
+        ['Connection'] = nil,
+        ['Accept'] = nil,
+    }
+    local r, _ = client.post(url, body, opts)
+    t.assert_equals(r.status, 200, 'HTTP code is not 200')
+    t.assert_equals(r.body, json.encode(body))
```

Comment:
Nit: Should not we also check the content type?

Reply:
Should. Added assert() for that.